backend/.env.sample

```diff
@@ -4,8 +4,14 @@ BIND_ADDRESS_HOST=127.0.0.1
 BIND_ADDRESS_PORT=8080
 
 # OAuth2 (other URLs will be fetched from this URL)
-AUTH_DISCOVERY_URI=https://auth.permaplant.net/realms/PermaplanT/.well-known/openid-configuration
+AUTH_HOST=https://auth.permaplant.net
 AUTH_CLIENT_ID=localhost
 
+# Keycloak API (will be used to fetch user info)
+KEYCLOAK_CLIENT_ID=admin-cli
+KEYCLOAK_CLIENT_SECRET=configured-in-keycloak
+KEYCLOAK_USERNAME=configured-in-keycloak
+KEYCLOAK_PASSWORD=configured-in-keycloak
+
```

Comment:
> Can you please add links to which config options these refer to in keycloak?

I will try to add links here and in the docs. I did not really rely on any docs from keycloak but configured the `admin-cli` user in the keycloak ui.

> Why is this password etc. needed if there is already a secret? Is there maybe an API key or similar which could simplify configuration?

I found `https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/admin-cli.html` maybe this can be used to infer a simpler configuration. I have to try it.

>Any specific reason why we prefix with KEYCLOAK? Is it more keycloak-specific than AUTH_HOST? What about following variable names:

Yes it is more specific, the keycloak admin rest api is something only provided by keycloak.
I guess we should rename the AUTH_HOST option to KEYCLOAK_HOST then. Our collaborator api would need to be rewritten if we chose to not use keycloak anymore.

Reply:
> I will try to add links here and in the docs. I did not really rely on any docs from keycloak but configured the admin-cli user in the keycloak ui.

Would be great to have a description how to do this, ideally even via keycloak API call so that we can simply execute this on our servers?

> I found https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/admin-cli.html maybe this can be used to infer a simpler configuration. I have to try it.

Set yourself a timeout for half an hour or less for simplifying the setup. More important is actually something else:

Is it possible to restrict this admin access to user discovery? (And disable dangerous operations like removal of users etc.?) If not, we probably need some way to disable the whole discovery feature in the backend (e.g. if auth admin env vars are not set). This would enable devs to continue developing on everything but user discovery without needing any setup.

> I guess we should rename the AUTH_HOST option to KEYCLOAK_HOST then. Our collaborator api would need to be rewritten if we chose to not use keycloak anymore.

Yes, I think it will be more consistent if we use KEYCLOAK_ or AUTH_ everywhere (I think it is fine that in our case the auth server is keycloak) .

Reply:
> Is it possible to restrict this admin access to user discovery? (And disable dangerous operations like removal of users etc.?) If not, we probably need some way to disable the whole discovery feature in the backend (e.g. if auth admin env vars are not set). This would enable devs to continue developing on everything but user discovery without needing any setup.

Do you mean in case of a malicious dev using the auth as a keycloak instance instead of a local one?
I think it is possible either way. The collaborator api needs to somehow have two implementations depending on whether the variables are set or not.

Reply:
> Do you mean in case of a malicious dev using the auth as a keycloak instance instead of a local one?

Currently all the devs I know about use auth.permaplant.net directly. We should avoid that everyone needs to install their own keycloak.

> I think it is possible either way. The collaborator api needs to somehow have two implementations depending on whether the variables are set or not.

I think that is the easiest and most developer-friendly variant.

If the vars can simply be ignored by devs (by not setting them), then it is also not really important if they are simple to set (as long as @lukashartl  manages).

The cli-admin on our Keycloak instance should get minimal rights either way, even if we don't pass along the credentials to devs.

> I guess rust traits can do this, I need to research this.

KIS: A simple `if` or `match` should suffice, too. Simply return nothing, on local instances collaboration does not really make sense (except if we need it for E2E tests).